pkg/deploy/controller/deployment/controller.go

```diff
@@ -369,8 +369,9 @@ func (c *DeploymentController) cleanupDeployerPods(deployment *kapi.ReplicationC
 	}
 
 	cleanedAll := true
+	gracePeriod := int64(10)
 	for _, deployerPod := range deployerList.Items {
-		if err := c.pn.Pods(deployerPod.Namespace).Delete(deployerPod.Name, &kapi.DeleteOptions{}); err != nil && !kerrors.IsNotFound(err) {
+		if err := c.pn.Pods(deployerPod.Namespace).Delete(deployerPod.Name, &kapi.DeleteOptions{GracePeriodSeconds: &gracePeriod}); err != nil && !kerrors.IsNotFound(err) {
```

Comment:
Actually, the correct place to put this is where you create the deployer pod, not here.